src/js/leaflet.storage.js

```diff
@@ -377,6 +375,13 @@ L.Storage.Map.include({
         this.options.tilelayer = L.extend({}, this._backupOptions.tilelayer);
     },
 
+    loadOptions: function () {
+        this.name = this.options.name;
+        this.description = this.options.description;
+        this.demoTileInfos = this.options.demoTileInfos;
+        
```

Comment:
In the end, I made it load the options by saving and refreshing the page. I didn't want my first patch to contain so much refactoring. :) I'll revert those few lines.